File: 1029-two-city-scheduling/1029-two-city-scheduling.java
class Solution {
    public int twoCitySchedCost(int[][] costs) {

        List<int[]> list = new ArrayList<>();
        int ans=0;

        for(int i=0; i<costs.length; i++){
            int diff = costs[i][0]-costs[i][1];
            list.add(new int[]{diff, i});
        }

        Collections.sort(list,(a,b)-> {
            if(a[0]==b[0]){
                return a[1]-b[1];
            }
            return a[0]-b[0];
        });
        
        int n=costs.length/2;

        for(int i=0; i<costs.length; i++){
             int val=list.get(i)[1];
             if(i<n){
             ans += costs[val][0];
             }else{
             ans += costs[val][1];
             }
        }

        return ans;
    }
}

File: 1029-two-city-scheduling/1029-two-city-scheduling.js
/**
 * @param {number[][]} costs
 * @return {number}
 */
var twoCitySchedCost = function(costs) {

    let ans=0;
    for(let i=0; i<costs.length ;i++){
        let val= costs[i][0]- costs[i][1]
        costs[i].push(val)
    }

    costs.sort((a,b)=> a[2] - b[2]);

    let n=costs.length;

    for(let i=0; i<n ; i++){

            if(i< n/2){
                ans+= costs[i][0]
            }else{
                ans+= costs[i][1]

            }
    }

    return ans;
    
};